core/src/test/scala/org/apache/spark/sql/store/TokenizationTest.scala

```diff
@@ -305,6 +305,19 @@ class TokenizationTest
       var res2 = snc.sql(query).collect()
       assert( cacheMap.size() == 4)
 
+      // test fetch first syntax also
+      // Cache map size doesn't change because fetch first converts into limit
+      // itself and the exact same thing is fired above
+      query = s"select * from $table where a = 0 fetch first 1 row only"
+      res1 = snc.sql(query).collect()
+      assert( cacheMap.size() == 4)
+
+      // Cache map size doesn't change because fetch first converts into limit
+      // itself and the exact same thing is fired above
+      query = s"select * from $table where a = 0 fetch first 10 rows only"
```

Comment:
Query can have order by and value of the rows returned should be verified.

Reply:
Done.